docs/user-guide/jobs.md

```diff
@@ -226,6 +228,113 @@ sometimes be started twice.
+| [Job Template Expansion](../../examples/job/expansions/README.md)          |          1          |     should be 1      |
+| [Queue with Pod Per Work Item](../../examples/job/work-queue-1/README.md)  |          W          |        any           |
+| [Queue with Variable Pod Count](../../examples/job/work-queue-2/README.md) |          1          |        any           |
+| Single Job with Static Work Assignment                                     |          W          |        any           |
+
+<!--
+When you specify completions with `.spec.completions`, each Pod created by the Job controller
+has an identical [`spec`](../devel/api-conventions.md#spec-and-status).  This means that
+all pods will have the same command line and the same
+image, the same volumes, and (almost) the same environment variables.  It is up
+to the user to arrange for the pods to do work on different things.  There are
+several different patterns for parallel jobs, each with strengths and weaknesses.
+The Job object supports   For
+example, the pods might all access a shared work queue service to acquire work
+units.
+-->
+
+### Work Queue with Pod Per Item
+
+The *multiple jobs from template* approach creates one job object for each work item, and so may be
+awkward when there are a very large number of work items.
+
+Also, sometimes you may want to limit the number of pods that are created at the same time, to
+conserve resources, and have the job run until items are completed.
+
+The work-queue-based approaches in this and the following section allow a single Job
+object to manage multiple pods running in parallel, and the number of completions to be
+dynamically adjusted at runtime.
+
+**TODO: write this example.  Also see if I can do it in a way that
+  there is a driver wrapper that invokes the program (e.g. with amqp-consume)**
+
+### Work Queue with Multiple Items per Pod
+
+If there are a large number of very short work items then it is inefficient to start a new pod for
+each work item, and there may be an excessive number of pods.
+
+In that case, each pod should process multiple work items.
+
+**TODO: write this example**
+**set num_completions to 1.**
+**Write the Pod so it never exits with success until it confirms that the workqueue is empty.**
+
+
+### Single Job with Static Work Assignment
+
+
+We plan to implement an approach where a list of parameters can be passed to each Pod via the
+downward API.
+
+This approach creates a pod per work item, but does not require modification of the program in the
+container image to be work-queue aware.
+
+### Single Job starts Controller Pod
+
+Another pattern is for a single Job to create a pod which then creates other pods, acting as a sort
+of custom controller for those pods.  This allows the most flexibility, but may be somewhat
+complicated to get started with and offers less integration with Kubernetes.
```

Comment:
I've meant the usual problems one have to face when implementing controller in k8s. This is sort of controller, although running differently, but the overlapping pods is one example one should be aware of.